### PR TITLE
Add labs.quansight.com -> .org redirect

### LINKED ISSUE #410: [FEAT] - Add redirect of `labs.quansight.com` to `labs.quansight.org`
### What site is this for?

Quansight LLC

### Expected behavior

_No response_

### Actual behavior

This issue serves to document what was done to achieve a redirect of `labs.quansight.com` to `labs.quansight.org`.

The first attempt was the NON CODE CHANGE of adding a CNAME entry to the Vercel DNS records for `quansight.com`:

- NAME: labs
- TYPE: CNAME
- VALUE: labs.quansight.org
- TTL: 3600
- PRIORITY: (empty)

This does not seem to have worked, as the `*` CNAME entry that Vercel uses to collect all incoming traffic to all subdomains appears to be taking precedence.

The solution instead may be a [Next.js redirect using `has.host`](https://stackoverflow.com/a/67211550/4376000) to select on the `labs` subdomain.

### How to Reproduce the problem?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by bskinn**
Attempting to blanket-redirect all incoming paths to .org

Closes #410.

### CODE REVIEW

**Comment by bskinn** [comment]
Need your help here, @gabalafou -- I tried to set up the redirect so that it would trigger even with the Vercel preview URLs, but it doesn't seem to be working right. When I prepend `labs.` to the preview URL, it complains of an SSL error.

I tried creating a static Vercel URL for the PR branch, https://quansight-redirect-test.vercel.app/, but the same thing still happens at https://labs.quansight-redirect-test.vercel.app/.

Do you have any thoughts on what's going on, and how to successfully test this before a merge to `develop` or `main`?

**Comment by bskinn** [comment]
I think I got it. I had to create Vercel deployment domains on the `quansight-consulting-live` project at *both* https://quansight-redirect-test.vercel.app *and* https://labs-quansight-redirect-test.vercel.app. This provided a valid build of the site at a host starting with `labs` (satisfying the `^labs[.-].+` regex) to test the redirect, and one *not* starting with `labs` to serve the actual LLC site.

Please check those two links, @gabalafou, and see if you think this is a solid implementation.

If so, once the change is merged to `main`, we will need to create the `labs.quansight.com` domain for the `quansight-consulting-live` project, in order for the redirect to work on the live site.

**Comment by bskinn** [comment]
The URL `:path` appears to be successfully transferred across the redirect, too; see e.g.:  https://labs-quansight-redirect-test.vercel.app/blog/the-evolution-of-the-scipy-developer-cli

**Comment by noatamir** [comment]
Thanks Brian! LGTM, but I would prefer @gabalafou double checks.

File: apps/consulting/next.config.js [hunk 1]
```diff
@@ -29,6 +29,17 @@ const nextConfig = {
         permanent: false,
         destination: '/careers',
       },
+      {
+        source: '/:path*',
+        has: [
+          {
+            type: 'host',
+            value: '^labs[.-].+$',
```
Comment (gabalafou) [inline apps/consulting/next.config.js:37]: Could you add a line explaining this regex? For example, why are you matching "labs" at the beginning of the string, but then anything at the end?

**Review by gabalafou — APPROVED** [review]
A code comment for the regular expression would be nice, but otherwise, this looks good to me.

File: apps/consulting/next.config.js (hunk 1 above)
Comment (bskinn) [reply apps/consulting/next.config.js:37]: Sure, will do. The terminating `$` is not necessary, I'll remove for simplicity.

Comment (bskinn) [reply apps/consulting/next.config.js]: Ok, comment added, thoughts pls @gabalafou.

**Review by gabalafou — APPROVED** [review]

File: apps/consulting/next.config.js (hunk 1 above)
Comment (gabalafou) [reply apps/consulting/next.config.js]: Looks good, thanks!

**Pull request merged**